src/inverted-index.js

```diff
@@ -1,49 +1,86 @@
 // Index object
 function Index() {
-  var book = [
-    {
-      "title": "Alice in Wonderland",
-      "text": "Alice falls into a rabbit hole and enters a world full of imagination."
-    },
-
-    {
-      "title": "The Lord of the Rings: The Fellowship of the Ring.",
-      "text": "An unusual alliance of man, elf, dwarf, wizard and hobbit seek to destroy a powerful ring."
-    }
+  this.fileContents = undefined;
+  this.invertedIndex = {};
+  // stop words to ignore on index creation
+  this.stopWords = ["a", "an", "and", "as", "at", "but", "by", "each", "every", "for",
+    "from", "her", "his", "in", "into", "its", "like", "my", "no", "nor",
+    "of", "off", "on", "onto", "or", "our", "out", "outside", "over", "past",
+    "since", "so", "some", "than", "that", "the", "their", "this", "to", "up", "with"
   ];
 
-  this.fileContents = [];
-  this.index_ = {};
-  // this.createIndex = function(filepath) {
-  //   fetch(filepath)
-  //     .then(function(response) {
-  //       response.json();
-  //       return response.json();
-  //     });
-  // };
+  this.readFile = function (filepath) {
+    var self = this;
+    return fetch(filepath)
+      .then(function (response) {
+        return response.text();
+      }).then(function (response) {
+        try {
+          self.fileContents = JSON.parse(response);
+        } catch (e) {
+          console.log(e);
+        }
+        return response;
+      }).catch(function (err) {
+        console.log('parsing failed', err);
+        throw err;
+      });
+  };
+
+  this.getIndex = function () {
+    var self = this;
 
-  this.getIndex = function() {
-     var self = this;
-     for (var i = 0; i < book.length; i++) {
-       var obj = book[i];
-       Object.keys(obj).map(function(key) {
-         self.fileContents.push(obj[key]);
-       });
-       strFileContents = this.fileContents.toString().toLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]+/gi, " ").replace(/\s{2,}/g, " ").trim().split(" ");
-      //  console.log(i);
-       for (var x = 0; x < strFileContents.length; x++) {
-         this.index_[strFileContents[x]] = i;
+    this.fileContents.forEach(function (element, index) {
+      for (var value in element) {
+        var strFileContents = element[value].toLowerCase()
+          .replace(/\W/gi, " ").replace(/\s{2,}/g, " ")
```

Comment:
**Code Styling - Critical**

- Use single quotes.
- the flag /i is unnecessary here
- You can replace the two separate regex with a single one like so:

```javascript
var str = element[value].toLowerCase()
   .replace(/\W+/g, ' ')
   .trim()
   .split(' ')